src/jdk.jfr/share/classes/jdk/jfr/internal/Utils.java

```diff
@@ -719,6 +725,20 @@ public static String formatDuration(Duration d) {
         }
     }
 
+    public static boolean shouldSkipBytecode(String eventName, Class<?> superClass) {
+        if (!superClass.getName().equals("jdk.jfr.events.AbstractJDKEvent")) {
```

Comment:
Was there a problem checking against the class instance? If so, perhaps you could add a check that the class is in the boot class loader (null).

Reply:
Yes, `AbstractJDKEvent` is package private so it is not accessible from here.